### ISSUE TITLE
Create a Java source code level downgrader

### ISSUE THREAD

**Issue by matozoid** [javaparser/javaparser#870]
Sometimes people are forced to use an older version of Java. That means that more and more libraries will become unusable as they upgrade. Wouldn't it be nice to have a tool that replaces all the newer language features with older ones? It could remove annotations, replace for-each with explicit iterators, change lambda's into anonymous class syntax, etc.

**Comment by klaushauschild1984** [comment]
Is there some meta model to determine the language level for every node?

**Comment by matozoid** [comment]
Nope, language changes are more complex than that. We simply parse a wide definition of the Java rules so that all Java levels parse. In the last few weeks we have started work on adding semantic validation on Java version by using the new validator framework. Not much there yet, though.

Note that to do this correctly, you'll spend months searching through the JLS, finding all kinds of obscure edge features and thinking up horrible workarounds. How would a lambda look in Java 1.0? What do we do with annotations when they are no longer supported? Those kinds of things.
